File: python/cohort_metadata.py
# Example Python script to work with exported data
import pandas as pd
import json

# Load cohort metadata
def load_cohort_data(json_path):
    with open(json_path, 'r') as f:
        data = json.load(f)
    return pd.DataFrame(data['cohort_metadata'])

# Filter images for analysis
def filter_cohort(df, criteria):
    filtered = df.copy()
    for column, value in criteria.items():
        if isinstance(value, tuple):  # Range filtering
            filtered = filtered[
                (filtered[column] >= value[0]) & 
                (filtered[column] <= value[1])
            ]
        else:
            filtered = filtered[filtered[column] == value]
    return filtered

# Example usage
cohort_df = load_cohort_data('cohort_metadata.json')
high_mag_images = filter_cohort(cohort_df, {
    'estimated_magnification': (20, 40),
    'area_mm2': (10, None)  # Minimum 10 mm²
})

File: R/cohort_metadata.R
# R script for cohort analysis
library(jsonlite)
library(dplyr)

# Load cohort data
load_cohort_data <- function(json_path) {
    data <- fromJSON(json_path)
    return(as.data.frame(data$cohort_metadata))
}

# Statistical analysis
analyze_cohort <- function(cohort_df) {
    summary_stats <- cohort_df %>%
        summarise(
            total_images = n(),
            mean_area_mm2 = mean(area_mm2, na.rm = TRUE),
            mean_magnification = mean(estimated_magnification, na.rm = TRUE),
            scanners = n_distinct(scanner, na.rm = TRUE)
        )
    return(summary_stats)
}
